controller/scrape.go

```diff
@@ -40,66 +41,104 @@ var numericStatus = map[string]float64{
-	engine, err := clientSet.LitmuschaosV1alpha1().ChaosEngines(exporterSpec.AppNS).Get(exporterSpec.ChaosEngine, metav1.GetOptions{})
+func GetLitmusChaosMetrics(clientSet *clientV1alpha1.Clientset) error {
+	allChaosEngineList, err := clientSet.LitmuschaosV1alpha1().ChaosEngines("").List(metav1.ListOptions{})
 	if err != nil {
-		return 0, 0, 0, nil, err
+		return err
 	}
-
-	for _, element := range engine.Spec.Experiments {
-		chaosExperimentList = append(chaosExperimentList, element.Name)
+	chaosEngineList := filterMonitoringEnabledEngines(allChaosEngineList)
+	if err != nil {
+		return err
 	}
+	var total, pass, fail float64
+	total = 0
+	pass = 0
+	fail = 0
+	for _, chaosEngine := range chaosEngineList.Items {
+		totalEngine, passedEngine, failedEngine, awaitedEngine := getExperimentMetricsFromEngine(&chaosEngine)
+		klog.V(2).Infof("ChaosEngineMetrics: EngineName: %v, EngineNamespace: %v, TotalExp: %v, PassedExp: %v, FailedExp: %v", chaosEngine.Name, chaosEngine.Namespace, totalEngine, passedEngine, failedEngine)
+		var engineDetails ChaosEngineDetail
+		engineDetails.Name = chaosEngine.Name
+		engineDetails.Namespace = chaosEngine.Namespace
+		engineDetails.TotalExp = totalEngine
+		engineDetails.PassedExp = passedEngine
+		engineDetails.FailedExp = failedEngine
+		engineDetails.AwaitedExp = awaitedEngine
+		total += totalEngine
+		pass += passedEngine
+		fail += failedEngine
+		setEngineChaosMetrics(engineDetails)
+	}
+	setClusterChaosMetrics(total, pass, fail)
+	time.Sleep(1000 * time.Millisecond)
+	return nil
+}
+
+func setClusterChaosMetrics(total float64, pass float64, fail float64) {
+	ClusterPassedExperiments.WithLabelValues().Set(pass)
+	ClusterFailedExperiments.WithLabelValues().Set(fail)
+	ClusterTotalExperiments.WithLabelValues().Set(total)
+}
+func setEngineChaosMetrics(engineDetails ChaosEngineDetail) {
+	EngineTotalExperiments.WithLabelValues(engineDetails.Namespace, engineDetails.Name).Set(engineDetails.TotalExp)
+	EnginePassedExperiments.WithLabelValues(engineDetails.Namespace, engineDetails.Name).Set(engineDetails.PassedExp)
+	EngineFailedExperiments.WithLabelValues(engineDetails.Namespace, engineDetails.Name).Set(engineDetails.FailedExp)
+	EngineAwaitedExperiments.WithLabelValues(engineDetails.Namespace, engineDetails.Name).Set(engineDetails.AwaitedExp)
+}
 
-	// Set default values on the chaosResult map before populating w/ actual values
-	spec := ChaosResultSpec{ExporterSpec: exporterSpec, ChaosExperimentList: chaosExperimentList}
-	chaosResultMap := setChaosResultValue(clientSet, spec)
+func getExperimentMetricsFromEngine(chaosEngine *litmuschaosv1alpha1.ChaosEngine) (float64, float64, float64, float64) {
+	var total, passed, failed, awaited float64
+	passed = 0
+	failed = 0
+	awaited = 0
+	expStatusList := chaosEngine.Status.Experiments
+	total = float64(len(expStatusList))
+	for i, v := range expStatusList {
+		verdictFloat := getValueFromVerdict(strings.ToLower(v.Verdict))
+		if verdictFloat == 4 {
```

Comment:
can we have it the other way round in terms of order: 

`awaited: 1`
`running: 2`
`failed: 3`
`passed: 4`

Makes a successful experiment the highest value on dashboards.